plugin/src/main/kotlin/com/vanniktech/maven/publish/tasks/JavadocJar.kt

```diff
@@ -14,33 +14,40 @@ open class JavadocJar : Jar() {
   }
 
   internal companion object {
-    internal fun Project.javadocJarTask(javadocJar: JavadocJarOption): TaskProvider<*>? {
+    internal fun Project.javadocJarTask(prefix: String, javadocJar: JavadocJarOption): TaskProvider<*>? {
       return when (javadocJar) {
         is JavadocJarOption.None -> null
-        is JavadocJarOption.Empty -> emptyJavadocJar()
-        is JavadocJarOption.Javadoc -> plainJavadocJar()
-        is JavadocJarOption.Dokka -> dokkaJavadocJar(javadocJar.taskName)
+        is JavadocJarOption.Empty -> emptyJavadocJar(prefix)
+        is JavadocJarOption.Javadoc -> plainJavadocJar(prefix)
+        is JavadocJarOption.Dokka -> dokkaJavadocJar(prefix, javadocJar.taskName)
       }
     }
 
-    private fun Project.emptyJavadocJar(): TaskProvider<*> = tasks.register("emptyJavadocJar", JavadocJar::class.java)
+    private fun Project.emptyJavadocJar(prefix: String): TaskProvider<*> = tasks.register(
+      "${prefix}EmptyJavadocJar",
+      JavadocJar::class.java,
+    ) {
+      it.archiveBaseName.set("$prefix-javadoc")
```

Comment:
Need to set this so that the output has different names per publication, this is just the local file name and doesn't influence how it is published.